reflectapi/src/codegen/mod.rs

```diff
@@ -25,3 +29,99 @@ fn tmp_path(src: &str) -> PathBuf {
 
     std::env::temp_dir().join(format!("reflectapi-{hash}"))
 }
+
+trait Subst {
```

Comment:
Is there better more meaningful name? Looking at it I do not understand what it does

Reply:
How about `Substitute` in full + some comments?

Reply:
Maybe. I still not quite understand what it does. Perhaps some examples would help?

Reply:
I've tried to add more comments, not finding it particularly easy to document `Substitute`. I've added more comments for `Instantiate` which provides the context for why substitution makes sense. Let me know if that makes sense.

Reply:
@avkonst are you sufficiently happy with this to merge?